worksheet_D/oxo.py

```diff
@@ -1,88 +1,176 @@
-                print "--+---+--"
-            for x in xrange(3):
-                if x > 0:
-                    print '|',
-
-                # Print a space for empty (0), an O for player 1, or an X for player 2
-                print " OX"[self.get_square(x, y)],
-            print
-
-
-def input_square():
-    """ Prompt the player to enter a square. You should not need to edit this. """
-    while True:
-        input = raw_input("Enter x,y where x=0,1,2, y=0,1,2: ")
-        if input.count(',') != 1:
-            print "Input must contain exactly one comma!"
-            continue
-
-        x, y = input.split(',')
-        try:
-            x = int(x)
-            y = int(y)
-        except ValueError:
-            print "Input must be two numbers separated by a comma!"
-            continue
-
-        if x < 0 or x > 2 or y < 0 or y > 2:
-            print "Input is out of bounds!"
-            continue
-
-        return x, y
-
-
-# The main game. You should not need to edit this.
-if __name__ == '__main__':
-    board = OxoBoard()
-    current_player = 1
-    while True:
-        board.show()
-        print "Choose a square, Player", current_player
-        x, y = input_square()
-
-        if board.set_square(x, y, current_player):
-            # Move was played successfully, so check for a winner
-            winner = board.get_winner()
-            if winner != 0:
-                print "Player", winner, "wins!"
-                break   # End the game
-            elif board.is_board_full():
-                print "It's a draw!"
-                break   # End the game
-            else:
-                # Switch players
-                if current_player == 1:
-                    current_player = 2
-                else:
-                    current_player = 1
-        else:
-            # Move was not played successfully
-            print "That square is already filled!"
+class OxoBoard:
+    def __init__(self, x, y):
+
+        # create a 2D array of a 3 X 3 grid
+        self.board = [[0, 0, 0],[0, 0, 0],[0, 0, 0]]
+
+    def get_square(self, x, y):
+        #this function returns the value of a square in the array
+        return self.board[x][y]
+
+
+    def set_square(self, x, y, mark):
+        #This function sets a square in the array
+        if self.get_square(x,y) == 0:
+            self.board[x][y] = mark
+            return True
+        else:
+            return False
+
+
+    def is_board_full(self):
+
+        fullTiles = 0
+
+        for i in range(len(self.board)):
+            for j in range(len(self.board[i])):
+                squareContent = self.get_square(i, j)
+                if squareContent != 0:
+                    fullTiles += 1
+                    if fullTiles == 9:
+                        return True
+                else:
+                    return False
+
+
+
+    def get_winner(self, latestX, latestY, currentPlayer):
+
+        total = latestX + latestY
+        checkDiagnols = 0

```

Comment:
You seem to be using this as a boolean flag, so better to use values of `False` and `True` rather than `0` and `1`

Also you have misspelled "diagonal"